Reviewing getting started guides (#10979)
DataDog/documentation: content/en/getting_started/synthetics/browser_test.md

    - link: '/getting_started/synthetics/private_location'
      tag: 'Documentation'
      text: 'Learn about private locations'
    - link: '/synthetics/identify_synthetics_bots'
      tag: 'Documentation'
      text: 'Learn how to identify Synthetic bots for API tests'
      
---

## Create a browser test

> [Browser tests][1] are scenarios that Datadog executes on your web applications. You can configure periodic intervals to run tests from multiple locations, devices, and browsers as well as execute them from your CI/CD pipelines. These tests verify that your users can perform key business transactions on your applications and that they are not badly impacted by the most recent code changes.

Reviewer: ```suggestion
[Browser tests][1] are scenarios that Datadog executes on your web applications. You can configure periodic intervals to run tests from multiple locations, devices, and browsers as well as execute them from your CI/CD pipelines. These tests verify that your users can perform key business transactions on your applications and that they are not negatively impacted by the most recent code changes.
```

2. In the top right corner, click the **New Test** button.
3. Select **Browser Test**.
4. Define your browser test:

    - Add the URL of the website you want to monitor. If you don’t know what to start with, you can use `https://www.shopist.io`, a test web application.
    - Select **Advanced Options** to use custom request headers, authentication credentials, or cookies. 
    - Name your test and set tags to it such as `env:prod` and `app:shopist`. Tags allow you to keep your test suite organized and quickly find tests you're interested in on the homepage.
    - Choose the browsers and devices you want to test with. 

#### Select locations

Select one or more **Managed Locations** or **Private Locations** to run your test from.

Managed locations allow you to test public-facing websites and endpoints. To test internal applications or simulate user behavior in discrete geographic regions, select one of your **Private Locations** instead.

For more information on how to set up private locations, see [Getting Started with Private Locations][3].

#### Specify test frequency

Select the frequency that your test runs.

#### Define alert conditions

> You can define alert conditions to ensure your test does not trigger in case of sporadic network blip and that you only get alerted in case of real issues with your application.

Reviewer: ```suggestion
You can define alert conditions to ensure your test does not trigger for things like a sporadic network blip, so that you only get alerted in case of real issues with your endpoint.
```

You can specify the number of consecutive failures that should happen before considering a location failed:

```text
Retry test 2 times after 300 ms in case of failure
```

> You can also decide to only trigger a notification when your endpoint goes down for a certain amount of time and number of locations. In the below example, we set our alerting rule to send a notification if our test fails for three minutes on two different locations:

Reviewer: ```suggestion
You can also define it to only trigger a notification when your endpoint goes down for a certain amount of time and number of locations. In the below example, the alerting rule is set to send a notification if the test fails for three minutes on two different locations:
```

```text
An alert is triggered if your test fails for 3 minutes from any 2 of 13 locations
```

#### Notify your team

Add an alert name to the **Monitor Name** field and write a message for the alert. You can use [integrations][4] such as Slack, PagerDuty, and webhooks to route your alert to specific services and teams.

> You can set time for your alert notification to re-notify if the alert has not been resolved and define the priority of the alert, ranging from **P5 (Info)** to **P1 (Critical)**.

Reviewer: ```suggestion
You can set your alert notification to re-notify if the alert has not been resolved and define the priority of the alert, ranging from **P5 (Info)** to **P1 (Critical)**.
```

1. Navigate to one of the furniture sections on the example website such as **Chairs** and select **Add to cart**.
2. Click on **Cart** and **Checkout**.
3. Under **Add New**, select **Assertion** and click **“Test that some text is present on the active page”**.

> 4. To confirm the words “Thank you!” appear after checking out, enter `Thank you!` in the **Value** field. 

Reviewer: ```suggestion
4. To confirm that the words “Thank you!” appear after checking out, enter `Thank you!` in the **Value** field. 
```

5. Press **Save & Quit**.

> It is important to finish your browser test with an **Assertion** as that is what ensures your application ended up in the expected state after the defined user journey.

Reviewer: ```suggestion
It is important to finish your browser test with an **Assertion** to ensure your application resulted in the expected state after the defined user journey.
```

{{< img src="getting_started/synthetics/record-test.gif" alt="Record test steps"  style="width:90%;">}}

**Note**: The example website regularly throws an error causing it to intentionally fail. If you include your email address in the **Notify your team** field, you will receive an email notification when the test fails and recovers.

## Test results

The Browser Test details page displays an overview of your test configuration, the global and per location uptime, graphs about time-to-interactive and test duration, sample successful and failed test results, and the list of all test results. Depending on the length of your test, you might have to wait for a few minutes to see the first test results come in.

> To troubleshoot a [failed test][6], select a failed test result and start by reviewing screenshots, including the ones leading up to the failed step. You can also review potential **[Errors & Warnings][7]**, **[Resources][8]** as well as **[Core Web Vitals][9]** to diagnose the issue. With Datadog's [APM integration with Synthetic Monitoring][10], you also have the ability to look into the traces that are generated in your backend by the test runs via the **Traces** tab.

Reviewer: ```suggestion
To troubleshoot a [failed test][6], select a failed test result and start by reviewing screenshots, including the ones leading up to the failed step. You can also review potential **[Errors & Warnings][7]**, **[Resources][8]** as well as **[Core Web Vitals][9]** to diagnose the issue. With Datadog's [APM integration with Synthetic Monitoring][10], you also have the ability to look into the traces that are generated in your backend by the test runs from the **Traces** tab.
```

In the example below, the test failed as the result of a server timeout.

{{< img src="getting_started/synthetics/browser-test-failure.png" alt="Browser test failure"  style="width:100%;">}}

> Thanks to the **Errors & Warnings** tab, you can see that the server responded with a `504` when the browser tried loading the `https://api.shopist.io/checkout.json` resource, which caused the browser test to fail.

Reviewer: ```suggestion
In this case, the **Errors & Warnings** tab displays that the server responded with a `504` when the browser tried loading the `https://api.shopist.io/checkout.json` resource, which caused the browser test to fail.
```
